source/diffHandler.py

```diff
@@ -21,25 +22,18 @@ class DiffAlgo(AutoPropertyObject):
 	def diff(self, newText: str, oldText: str) -> List[str]:
 		raise NotImplementedError
 
-	_abstract_unit = True
-
-	def _get_unit(self):
+	@abstractmethod
+	def _getText(self, ti: TextInfo) -> str:
 		raise NotImplementedError
 
-	def _get_name(self):
-		return "diffing algorithm"
-
-	def __repr__(self):
-		return f"{self.unit}-based ({self.name})"
-
 
 class DiffMatchPatch(DiffAlgo):
-	name = "diff match patch"
-	unit = UNIT_CHARACTER
-
-	def __init__(self, *args, **kwargs):
-		self._proc = None
-		return super().__init__(*args, **kwargs)
+	(
+		"A character-based diffing approach, using the Google Diff Match Patch "
+		"library in a proxy process (to work around a licence conflict)."
+	)

```

Comment:
Normally we use triple quote syntax. It would be good to be consistent, unless there is a reason not to be?

```suggestion
	"""A character-based diffing approach, using the Google Diff Match Patch
	library in a proxy process (to work around a licence conflict).
	"""
```

Reply:
This would leave tabs (indentation for the source) in the actual string literal.

Reply:
```python console
>>> a=(
... 	"this is "
... 	"a test."
... )
>>> b="""
... 	this is
... 	a test.
... """
>>> a
'this is a test.'
>>> b
'\n\tthis is\n\ta test.\n'
```